### PR TITLE
feat: add instance detail pages

### PR DESCRIPTION
**Pull request by Megrax**
Shot-2022-10-19-Microsoft Edge-8qsJ7BmP

### CODE REVIEW

**Comment by d-bytebase** [comment]
@tianzhou, would you help take a look too?

File: frontend/components/RelationTable.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import { Table } from "antd";
```
Comment (tianzhou) [inline frontend/components/RelationTable.tsx:1]: RelatedTable ?

File: frontend/pages/index.tsx [hunk 2]
```diff
@@ -1,14 +1,168 @@
...
+            dataRowMap.set(regionInstanceKey, [newRow]);
+          }
+        });
+      });
+
+      dataRowMap.forEach((rows) => {
+        rows.sort((a, b) => {
+          // Sort rows according to the following criterion:
+          // 1. On demand price goes first.
+          // 2. Sort on expected cost.
```
Comment (tianzhou) [inline frontend/pages/index.tsx:132]: ASC or DESC?

**Review by tianzhou — CHANGES_REQUESTED** [review]
LGTM overall. Some minor comments

**Comment by Megrax** [comment]
> CleanShot 2022-10-19 at 19 32 04
> 
> 1. Title just use "db-N1Standard-64-240 Pricing" since that's the one people search for.
> 2. The db-N1Standard-64-240 instance has a CPU count of 64 and memory count of 240 from GCP. -> The GCP db-N1Standard-64-24 has 64 CPU and 240 GB memory

Should the title of tab be the same as `db-N1Standard-64-240 Pricing`?
Shot-2022-10-19-Microsoft Edge-rFj40Zjv

**Comment by Megrax** [comment]
PTAL

File: frontend/components/RelatedTable.tsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+import TdCell from "@/components/TdCell";
+import type { RelatedType } from "@/types/table";
+
+interface Props {
+  title: string;
+  instance: string;
+  dataSource: RelatedType[];
+}
+
+const RelationTable: React.FC<Props> = ({ title, instance, dataSource }) => {
```
Comment (tianzhou) [inline frontend/components/RelatedTable.tsx:13]: Needs to change this as well

**Comment by tianzhou** [comment]
CleanShot 2022-10-19 at 22 37 02

No dash between instance type and "pricing"

**Review by tianzhou — CHANGES_REQUESTED** [review]

**Review by tianzhou — APPROVED** [review]
LGTM

**Pull request merged**